collapse/groupby-collapse.R

```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/env Rscript
+
+cat("# groupby-collapse.R\n")
+
+source("./_helpers/helpers.R")
+
+stopifnot(requireNamespace("data.table", quietly=TRUE)) # collapse does not support integer64. Oversized ints will be summed to double. 
+.libPaths("./collapse/r-collapse") # tidyverse/collapse#4641
+suppressPackageStartupMessages(library("collapse", lib.loc="./collapse/r-collapse", warn.conflicts=FALSE)) 
+ver = packageVersion("collapse")
+git = "" # uses stable version now #124
+task = "groupby"
+solution = "collapse"
+fun = "group_by"
+cache = TRUE
+on_disk = FALSE
+
+data_name = Sys.getenv("SRC_DATANAME")
+src_grp = file.path("data", paste(data_name, "csv", sep="."))
+cat(sprintf("loading dataset %s\n", data_name))
+
+x = data.table::fread(src_grp, showProgress=FALSE, stringsAsFactors=TRUE, na.strings="", data.table=FALSE)
+print(nrow(x))
+gc()
+
+big_data <- nrow(x) > 1e8
+
+# Setting collapse options: namespace masking and performance
+oldopts <- set_collapse(nthreads = data.table::getDTthreads(), 
```

Comment:
@vincentarelbundock @SebKrantz I now believe that this should be changed to `nthreads = parallel::detectCores()` for parity. See #47.

Reply:
Thanks. To be honest I think tough it won’t matter if it’s like 40 or 90 threads. Because grouping and joins in collapse is still fully serial, and all the basic operating are parallelized across columns. The only place where collapse can take advantage of the machine is computing the median over groups, but again I expect the grouping performance to dominate everything. Let’s see if collapse can still be competitive with some of those parallel beasts ;).